src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/HalfConverter.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class HalfConverter : JsonPrimitiveConverter<Half>
+    {
+        private const int MaxFormatLength = 16;
+        private const int MaxEscapedFormatLength = MaxFormatLength * JsonConstants.MaxExpansionFactorWhileEscaping;
```

Comment:
I'm not entirely what's a good number here since I couldn't find a good way to determine what's the max amount of bytes `Half.TryFormat` can use.

Reply:
Presumably you mean `Half.TryParse`? @tannergooding might know

Reply:
What are you trying to do here in particular? The underlying parsing algorithm needs to be able to track up to 20 significant digits (for `Half`, it's 113 for `Single`, and 768 for `Double`) to ensure a correct result.

However, the entire input string must always be passed in such that it can process non-significant digits (such as leading zeros) and all trailing digits so that it can determine if the rounding goes up or down.

Imagine for example if the user defines `000...005` or `0.500...1`, etc. All the zero digits that represented by the `...` must be processed to ensure the result is correct and to ensure the relevant end of string is located.

Reply:
Actually I suspect BitConverter.ToHalf doesn't have the TryParse wiggle room.

Reply:
That's not an equivalent API. `BitConverter.ToHalf` simply does a reinterpret cast of raw bytes into a `Half`.

You'd need to do something similar to the default interface implementation for `IUtf8SpanParsable` done by `INumberBase<T>` here: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs,555

Which is to say, you have to transcode the input string to UTF-16, then try to parse.